### PR TITLE
django-libsass

### PR DESCRIPTION
**Pull request by sannykr**

### CODE REVIEW

File: recipes/django-libsass/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  run:
+    - python
+    - django >=1.8
+    - django-compressor >=1.3
+    - libsass >=0.7.0
+
+test:
+  import:
+    - django_libsass
+about:
```
Comment (dougalsutherland) [inline recipes/django-libsass/meta.yaml]: Nitpicking: there should be a blank line before this one.

**Comment by dougalsutherland** [comment]
The CIs aren't actually running the import test, because the `libsass` runtime dep isn't yet available (https://github.com/conda-forge/staged-recipes/pull/3707), and apparently this causes `conda-build` to build the package but then say `nothing to test` because it's not installable. This is strange, and should probably cause the build to fail somehow....

**Comment by scopatz** [comment]
Closing and reopening to trigger rebuild.

**Comment by sannykr** [comment]
@dougalsutherland: In the setup.py file, libsass is listed as a dependency and it is not on conda-forge yet. Can you please tell me how it passed all checks.

**Comment by dougalsutherland** [comment]
Yeah, I don't understand it. Will look into tomorrow.

**Comment by dougalsutherland** [comment]
I'm closing/reopening this so it rebuilds now that `libsass` exists, though I'm still confused why it seemed to work before.

**Comment by dougalsutherland** [comment]
This could be `noarch: python` as well.

I also sent a PR to include the license in distributions: https://github.com/torchbox/django-libsass/pull/28.

**Comment by dougalsutherland** [comment]
CIs are still saying `Nothing to test for: /staged-recipes/build_artefacts/linux-64/django-libsass-0.7-py36_0.tar.bz2`.

And...aha, I figured out why! You wrote `test/import`, but it should be `test/imports`. :)

Guess it's a linter bug that it doesn't recognize that as a problem; will open an issue.

**Comment by sannykr** [comment]
In libsass also, I have written "import". Let me check if it found the tests.

File: recipes/django-libsass/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
+{% set name = "django-libsass" %}
+{% set version = "0.7" %}
+{% set sha256 = "49db3334b87e1f7955c4f9fb9945bc296f8bfd27a14d6d89706e4b0e5dc5de1c" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
```
Comment (dougalsutherland) [inline recipes/django-libsass/meta.yaml]: You can delete this line, it's not needed.

File: recipes/django-libsass/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  script: python setup.py install --single-version-externally-managed --record record.txt
+
```
Comment (dougalsutherland) [inline recipes/django-libsass/meta.yaml]: This package should be `noarch: python`.

File: recipes/django-libsass/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    - python
+    - setuptools
+  run:
+    - python
+    - django >=1.8
+    - django-compressor >=1.3
+    - libsass >=0.7.0
+
+test:
+  import:
```
Comment (dougalsutherland) [inline recipes/django-libsass/meta.yaml]: This still says `import` and should be `imports`.

File: recipes/django-libsass/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    - django_libsass
+about:
+  home: https://github.com/torchbox/django-libsass
+  license: BSD-3-Clause
+  license_family: BSD
+  license_file: '{{ environ["RECIPE_DIR"] }}/LICENSE'
```
Comment (dougalsutherland) [inline recipes/django-libsass/meta.yaml]: Can you add a comment linking to https://github.com/torchbox/django-libsass/pull/28?

File: recipes/django-libsass/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+  summary: 'A django-compressor filter to compile SASS files using libsass'
+  description: |
+    A django-compressor filter to compile Sass files using libsass.
+  doc_url: https://github.com/torchbox/django-libsass
+  dev_url: https://github.com/torchbox/django-libsass
+
+extra:
+  recipe-maintainers:
+    - cshaley
```
Comment (dougalsutherland) [inline recipes/django-libsass/meta.yaml]: @cshaley, you happy to be a maintainer here?

**Comment by dougalsutherland** [comment]
@sannykr I went ahead and made a few minor changes in https://github.com/conda-forge/staged-recipes/pull/3708/commits/f24cf41bf1bffe9f14608ff46f78dd9d5e758260. Assumings tests pass now that it's actually testing something, I think we just need confirmation from @cshaley and then we're good.

**Comment by dougalsutherland** [comment]
Apparently `django_libsass` isn't importable without establishing django settings.

I guess the way to work around this is to get rid of the `imports` test and instead have a `run_tests.py` file that first calls `django.settings.configure()` (or whatever, I don't remember how all that works) and then does `import django_libsass`.

**Comment by sannykr** [comment]
@dougalsutherland : Thanks. I will do that.

File: recipes/django-libsass/meta.yaml (hunk 6 above)
Comment (cshaley) [reply recipes/django-libsass/meta.yaml]: Yes

**Comment by cshaley** [comment]
@sannykr here's the bare minimum working test:
```
import django
from django.conf import settings
settings.configure(
	INSTALLED_APPS=['compressor'],
	STATIC_URL='/',
	STATIC_ROOT='',
    COMPRESS_PRECOMPILERS = (('text/x-scss', 'django_libsass.SassCompiler'),)
) 
django.setup() 
import django_libsass
```

**Comment by sannykr** [comment]
@conda-forge/staged-recipes : Could you please review and merge. It seems there is some issue with python 3.5 on windows and that's why only that check failed.

**Comment by dougalsutherland** [comment]
@sannykr The failure on Windows python 2.7 is legitimate:

```
The following specifications were found to be in conflict:
  - libsass >=0.7.0 -> python 3.5*
  - python 2.7.*
```

This is because `libsass` doesn't build on python 2.7 on Windows. Hence this recipe won't work there either, so you need to add <code>skip: true&nbsp;&nbsp;# [win and py27]</code>, and, unfortunately, that means it actually can't be `noarch: python` either.

**Comment by isuruf** [comment]
Actually, it can be `noarch: python`. On windows and py27, conda will give an error saying `libsass` doesn't build on py27.

**Comment by dougalsutherland** [comment]
@isuruf So should it be `noarch: python` with no `skip` statements, and it's fine because it still won't be installable when `libsass` isn't available? That makes sense I guess.

**Comment by jakirkham** [comment]
FWIW I can see both points here. On the one hand, it should be possible to install this everywhere, but Windows Python 2.7 is really a pain to get things to work on for reasons that extend beyond Python 2.7 (i.e. Visual Studio 2008 🙄). On the other hand, claiming this is a noarch package implies it should actually install and work on Windows Python 2.7. I think the cautious and more accurate approach would be to **not** use noarch and simply skip Windows on Python 2.7. However I think it is really up to the maintainers as they are the ones that may have to field requests for Windows Python 2.7.

**Comment by dougalsutherland** [comment]
@jakirkham, just to be clear: _this_ package is pure python and would work on Windows py27 if you somehow got a working `libsass` there. But `libsass` doesn't compile there for visual studio reasons. I'm coming around to @isuruf's suggestion of leaving it noarch: `conda` gives a reasonable error message, it's semantically correct, and if we ever patch `libsass` to work on win py27 it won't require any changes here.

**Comment by sannykr** [comment]
@conda-forge/staged-recipes : Could you please review and merge.

**Comment by dougalsutherland** [comment]
Merging this version that isn't noarch.

**Comment by jakirkham** [comment]
Definitely see and understand both arguments @dougalsutherland. As I was trying to say at the end, I'd prefer to leave the `noarch` decision up to the maintainers of this recipe.

**Pull request merged**